uaclient/contract.py

```diff
@@ -177,6 +177,22 @@ def report_machine_activity(self):
             machine_token["activityInfo"] = response
             self.cfg.write_cache("machine-token", machine_token)
 
+    def get_updated_contract_info(
+        self,
+        machine_token: str,
+        contract_id: str,
+        machine_id: str = None,
+        detach: bool = False,
+    ) -> Dict[str, Any]:
+        """Get the updated machine token from the contract server."""
+        data = self._get_platform_data(machine_id)
+        return self._get_updated_contract_info(
```

Comment:
I think instead of reusing this method, we can just GET `API_V1_TMPL_CONTEXT_MACHINE_TOKEN_RESOURCE` right here. That function already serves too many purposes IMO and adjusting it to support a "GET" would make it even more complex.

Reply:
@CalvoM Don't know if I am missing something, but do we still need to address this part here ?